packages/flutter/lib/src/material/badge.dart

```diff
@@ -190,7 +204,10 @@ class Badge extends StatelessWidget {
     final AlignmentGeometry effectiveAlignment = alignment ?? badgeTheme.alignment ?? defaults.alignment!;
     final TextDirection textDirection = Directionality.of(context);
     final Offset defaultOffset = textDirection == TextDirection.ltr ? const Offset(4, -4) : const Offset(-4, -4);
-    final Offset effectiveOffset = offset ?? badgeTheme.offset ?? defaultOffset;
+    // Adds a offset const Offset(0, 8) to avoiding breaking customers after
+    // the offset calculation changes.
+    // See https://github.com/flutter/flutter/pull/146853.
+    final Offset effectiveOffset = (offset ?? badgeTheme.offset ?? defaultOffset) + const Offset(0, 8);
```

Comment:
@QuncCccccc This is the best way I can think of to cause least breaking for people who uses Badge(offset).

The customer that have a big jump in icon
https://fusion2.corp.google.com/presubmit/625484245/OCL:625484245:BASE:625495583:1713311589857:35f5826b/targets/invocations/a540f325-e68b-49cd-bce2-dc65254fb25d/targets/%2F%2Fanalytics%2Fmobile%2Fgiant%2Fui%2Fscreens:screens_tests_test_intelligent_home_screen_scuba_test.host/tests
is because they have a child which is too big in size so it uses offset to align the badge.

Adding this constant will make sure the offset they passed adjust the badge the same way.